sw/device/tests/aon_timer_wdog_bite_reset_test.c

```diff
@@ -0,0 +1,104 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/freestanding/limits.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+#include "sw/device/lib/testing/test_framework/test_status.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+// TODO:(lowrisc/opentitan#XXXX) Move these difinitions to somewhere else?
+// Power on reset (POR).
+#define RSTMGR_REASON_POWERON (1 << 0)
+// Low power exit (LOW_POWER_EXIT).
+#define RSTMGR_REASON_LOWPOWER_EXIT (1 << 1)
+//  Non-debug module (NDM).
+#define RSTMGR_REASON_NON_DEBUG_MODULE (1 << 2)
+//  Software issued request (SW).
+#define RSTMGR_REASON_SOFTWARE_REQUEST (1 << 3)
+// Hardware requests (HW_REQ).
+#define RSTMGR_REASON_SYSRST_CTRL (1 << 4)
+#define RSTMGR_REASON_WATCHDOG (1 << 5)
+#define RSTMGR_REASON_POWER_UNSTABLE (1 << 6)
+#define RSTMGR_REASON_ESCALATION (1 << 7)
+
+const test_config_t kTestConfig;
+
+/**
+ * Execute the aon timer wdog bite reset test.
+ */
+static void execute_test(dif_aon_timer_t *aon_timer, dif_pwrmgr_t *pwrmgr,
+                         uint32_t bark_time_us) {
+  uint32_t bark_cycles = (bark_time_us * kClockFreqAonHz / 1000000);
```

Comment:
Should the intermediate product be stored in a `uint64_t`? I think bad things might happen if `bark_time_us` is big.